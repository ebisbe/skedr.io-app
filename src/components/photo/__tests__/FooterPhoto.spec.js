import { shallowMount } from '@vue/test-utils'
import Vue from 'vue'
import Vuetify from 'vuetify'
import Comp from '../FooterPhoto.vue'

Vue.use(Vuetify)

describe('FooterPhoto', () => {
  it('renders title and subtitle properties', () => {
    const wrapper = shallowMount(Comp, {
      propsData: { title: 'My title', subtitle: 'My subtitle' }
    })
    expect(wrapper).toMatchSnapshot()
  })

  it('renders title and subtitle slots', () => {
    const wrapper = shallowMount(Comp, {
      slots: {
        title: '<span>Slot title</span>',
        subtitle: '<span>Slot subtitle</span>',
        action: '<span>Slot action</span>'
      }
    })
    expect(wrapper).toMatchSnapshot()
  })
})
