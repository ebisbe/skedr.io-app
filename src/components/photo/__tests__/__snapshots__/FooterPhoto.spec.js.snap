// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FooterPhoto renders title and subtitle properties 1`] = `
<v-container-stub tag="div" class="footer">
  <v-list-stub tag="div" tile="true" twoline="true">
    <v-list-item-stub activeclass="" tag="div">
      <v-list-item-content-stub>
        <v-list-item-title-stub class="white--text">
          My title
        </v-list-item-title-stub>
        <v-list-item-subtitle-stub>
          My subtitle
        </v-list-item-subtitle-stub>
      </v-list-item-content-stub>
    </v-list-item-stub>
  </v-list-stub>
</v-container-stub>
`;

exports[`FooterPhoto renders title and subtitle slots 1`] = `
<v-container-stub tag="div" class="footer">
  <v-list-stub tag="div" tile="true" twoline="true">
    <v-list-item-stub activeclass="" tag="div">
      <v-list-item-content-stub>
        <v-list-item-title-stub class="white--text"><span>Slot title</span></v-list-item-title-stub> <span>Slot subtitle</span>
      </v-list-item-content-stub> <span>Slot action</span>
    </v-list-item-stub>
  </v-list-stub>
</v-container-stub>
`;
